frontend/src/components/Profile/Profile.tsx

```diff
@@ -32,122 +48,155 @@ const InnerProfile: React.FC = () => {
     setOpen(false);
   };
 
+  useEffect(() => {
+    if (user && user.user_metadata) {
+      setFormData((prev) => ({
+        ...prev,
+        ...user.user_metadata as UserMetadata,
+      }));
+    }
+    if (user && user.email) {
+      setFormData((prev) => ({ ...prev, email: user.email as string }));
+    }
+  }, [user]);
+
+  const handleSave = async () => {
+    if (user) {
```

Comment:
What's the reason for this user check here? It's not being used in the rest of the function.

Reply:
The if statement can be removed since we have default form for unlogged in state